### DISCUSSION TITLE
compile with openwrt22.03 and asterisk18 for MT7628AN

### DISCUSSION THREAD

**Discussion by zombob** [IchthysMaranatha/asterisk-chan-quectel#45] [Q&A] [answered]
Can you show the detail steps for compile process with openwrt22.03 and asterisk18 ？
I wana to run asterisk18 with chan_quectel on my MT7628AN router.
I have less experience working with this.
Thanks for help.
I can sponsor some expenses, if needed.

**Comment by IchthysMaranatha** [comment]
Very sorry, won't be able to guide you through exact steps, can provide some broad guidelines. 

For 22.03, you'll need to use this sdk to build the package https://downloads.openwrt.org/releases/22.03.0/targets/ramips/mt76x8/openwrt-sdk-22.03.0-ramips-mt76x8_gcc-11.2.0_musl.Linux-x86_64.tar.xz
You can use an ubuntu machine on the cloud also for this, for eg you can get a free Oracle cloud account and set up a vm for this.
You can pick up tips on how to use the sdk to build a package from here http://dvblog.soabit.com/building-custom-openwrt-packages-an-hopefully-complete-guide/
You can use the makefile for Asterisk 18 given here (do not forget to use the patches directory at the same level as makefile) (I haven't tested with Ast18 myself, I built for openwrt 19 with Asterisk 16)

**Reply by zombob** [reply]
Thanks , I'll try it the way you say

**Comment by IchthysMaranathaCopy** [comment]
Had some time, so built it for you, please install and check.
[chan_quectelOpenwrt22Ast18mips24kc.zip](https://github.com/IchthysMaranatha/asterisk-chan-quectel/files/11248976/chan_quectelOpenwrt22Ast18mips24kc.zip)
Use ipk or copy chan_quectel.so to usr/lib/asterisk/modules and quectel.conf to etc/asterisk on the router. Configure quectel.conf as needed and restart asterisk service (reboot in case of problems)

**Reply by zombob** [reply]
Thank you soooo much!
It works!
Please leave your account number, It's my pleasure to buy a few cups of coffee for you.

**Reply by IchthysMaranathaCopy** [reply, marked as answer]
Thank you for the kind offer, but I have all I need, thanks to Him. If you'd really like to, please contribute something here https://bethmyriam.org/

**Reply by maisx2021** [reply]
![image](https://github.com/IchthysMaranatha/asterisk-chan-quectel/assets/92543774/dcb6501d-3182-462d-95af-a54c77a412bd)
Hi everyone, I'm also trying to install your promising module on OpenWRT and Raspberry Pi. I would need help on how to generate the package for Asterisk 20. 
Thanks a lot to everyone. 
Giovanni.

**Reply by IchthysMaranatha** [reply]
Extremely late reply but this repo will help you generate any openwrt binary needed
https://github.com/IchthysMaranatha/chan-quectel-openwrt-generator

For the rpi4 provide sdk link https://downloads.openwrt.org/releases/23.05.3/targets/bcm27xx/bcm2711/openwrt-sdk-23.05.3-bcm27xx-bcm2711_gcc-12.3.0_musl.Linux-x86_64.tar.xz and choose Ast20 from dropdown and run the github action
